File: frontend/src/services/storyService.ts
import axios from "axios";

import { Story } from "../types/Story";



const API_URL = "http://localhost:8080/api";



export const getAllStories = async (): Promise<Story[]> => {

  const response = await axios.get(`${API_URL}/stories`);

  return response.data;

};



export const getStoriesByDepartment = async (department: string): Promise<Story[]> => {

  const response = await axios.get(`${API_URL}/stories/department/${department}`);

  return response.data;

};



export const getStoryById = async (id: number): Promise<Story> => {

  const response = await axios.get(`${API_URL}/stories/${id}`);

  return response.data;

};



export const createStory = async (story: Story): Promise<Story> => {

  const response = await axios.post(`${API_URL}/stories`, story);

  return response.data;

};



export const updateStory = async (id: number, story: Story): Promise<Story> => {

  const response = await axios.put(`${API_URL}/stories/${id}`, story);

  return response.data;

};



export const deleteStory = async (id: number): Promise<void> => {

  await axios.delete(`${API_URL}/stories/${id}`);

};

File: backend/src/main/java/com/via/tabloid/backend/service/StoryService.java
package com.via.tabloid.backend.service;



import com.via.tabloid.backend.model.Story;

import com.via.tabloid.backend.repository.StoryRepository;

import org.springframework.beans.factory.annotation.Autowired;

import org.springframework.stereotype.Service;

import java.util.List;



@Service

public class StoryService {

    

    private final StoryRepository storyRepository;

    

    @Autowired

    public StoryService(StoryRepository storyRepository) {

        this.storyRepository = storyRepository;

    }

    

    public List<Story> getAllStories() {

        return storyRepository.findAll();

    }

    

    public List<Story> getStoriesByDepartment(String department) {

        return storyRepository.findByDepartment(department);

    }

    

    public Story getStoryById(Long id) {

        return storyRepository.findById(id).orElse(null);

    }

    

    public Story createStory(Story story) {

        return storyRepository.save(story);

    }

    

    public Story updateStory(Long id, Story storyDetails) {

        Story story = storyRepository.findById(id).orElse(null);

        if (story != null) {

            story.setTitle(storyDetails.getTitle());

            story.setContent(storyDetails.getContent());

            story.setDepartment(storyDetails.getDepartment());

            return storyRepository.save(story);

        }

        return null;

    }

    

    public boolean deleteStory(Long id) {

        Story story = storyRepository.findById(id).orElse(null);

        if (story != null) {

            storyRepository.delete(story);

            return true;

        }

        return false;

    }

}
